rsts/concepts/lifecycle_workflow_with_plugin.rst

```diff
@@ -0,0 +1,252 @@
+.. _lifecycle-workflow:
+
+#################################################################
+Understand the Lifecycle of a Flyte Workflow
+#################################################################
+
+After reading this entry you should be able to understand how Flyte's Plugin machinery
+work under the hood and how information flows from the different
+components in Flyte.
+
+Under the hood Flyte relies on a primitive called “Plugins”. Every task
+that you run on Flyte is somehow powered by a plugin. Some of those plugins are native and guaranteed by Flyte system. These native plugins are for example powering your python functions (i.e: Tasks) by running them inside a k8s pod. There are Three native plugins : ``Container``, ``K8sPod`` and ``Sql``.
+
```

Comment:
Under the hood, Flyte relies on a primitive called “Plugins”. Every task that you run on Flyte is powered by a plugin. Some of these plugins are native and guaranteed by Flyte system. These native plugins, for example, run your Python functions (such as tasks) inside a k8s pod. There are three native plugins: ``Container``, ``K8sPod`` and ``Sql``.